docs/advanced/making plugins.md

```diff
@@ -297,3 +300,43 @@ Note that it takes in a `FullPageLayout` as the options. It's made by combining
 
 > [!hint]
 > Look in `quartz/plugins` for more examples of plugins in Quartz as reference for your own plugins!
+
+## quartz-plugin
+
+A plugin for use with quartz-plugin needs two things: a `plugin_name.ts` file and a `plugin.json`.
+
+The `plugin_name.ts` file will export a type that implements `QuartzTransformerPlugin`, `QuartzEmitterPlugin`, or `QuartzFilterPlugin` like explained above.
+
+`plugin.json` is the bridge between `plugin_name.ts` and the rest of Quartz. Here's an example `plugin.json` file:
+
+```json
```

Comment:
also imho json is just a horrible way to create and manage plugins. we can just have a `quartz.plugins.ts` top level for plugins dev to define whatever they need.

worth to also think about types for these options as well.

Reply:
To provide default configs and exports that align with Quartz's plugin architecture, I either need codegen or the plugin authors to tell me what to write. I opted for the latter, feel free to write the former based on the plugin file. 

As for the toml suggestion, I'm a huge rust fan but json is more aligned with web tech. Thoughts on swapping to yaml?

Reply:
I think we can export the types for what quartz expect from plugins dev.

We have full control of this. So you can write down any expectation you need from plugin dev such that you can just import the module as a node module.

Reply:
I can't inspect either of those at build time from a JavaScript handler:
1. Can't import typescript from JavaScript
2. Type information is completely discarded after compile time so I'd be losing the benefit of doing it in TS because it's now, like a json, just a key-value pair

Reply:
also see L559 of handlers, I wanted to inspect the array in plugins ts but will have to do so with regex because I can't import from that file.

Reply:
I think it is reasonable to ask plugin dev to write it in typescript right?

most components and emitters are currently structured in typescript. I'm sure we can figure out a way to export the type.

Also I'm not sure but does JSDoc allow you to import type from typescript package?

Then we can always ensure a type-safe package.